h2o-py/tests/testdir_algos/automl/target_encoding/pyunit_target_encoding_model.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import print_function
+import sys, os
+import tempfile
+
+sys.path.insert(1, os.path.join("..","..","..",".."))
+import h2o
+from tests import pyunit_utils
+from h2o.estimators import H2OTargetencoderEstimator
+
+"""
+This test is used to check Rapids wrapper for java TargetEncoder
+"""
+
+def test_target_encoding_fit_method():
+    print("Check fit method of the TargetEncoder class")
+    targetColumnName = "survived"
+    foldColumnName = "kfold_column"
+
+    teColumns = ["home.dest", "cabin", "embarked"]
+    trainingFrame = h2o.import_file(pyunit_utils.locate("smalldata/gbm_test/titanic.csv"), header=1)
+
+    trainingFrame[targetColumnName] = trainingFrame[targetColumnName].asfactor()
+    trainingFrame[foldColumnName] = trainingFrame.kfold_column(n_folds=5, seed=1234)
+    
+    te = H2OTargetencoderEstimator(encoded_columns = teColumns, target_column = targetColumnName, k = 0.7, f = 0.3, data_leakage_handling = "none")
+    te.train(training_frame = trainingFrame)
+    print(te)
+    transformed = te.transform(trainingFrame)
+    
+    assert transformed is not None
+    print(transformed.names)
+    assert transformed.ncols == trainingFrame.ncols + len(teColumns)
+    for te_col in teColumns:
+        assert te_col + "_te" in transformed.names
+    
+    assert transformed.nrows == 1309
+    
+    # Test fold_column proper handling + kfold data leakage strategy defined
+    te = H2OTargetencoderEstimator(encoded_columns = teColumns, target_column = targetColumnName, fold_column = "pclass",
```

Comment:
In regular estimators in Python we have "data related" arguments in the `.train()` method and not the Estimator constuctor (including exact parameters like `fold_column`.  Maybe that's something we want to follow here as well for some or all of the data related parameters?

Reply:
Thank you. Done.

Now, our users will be able to do:

```
    te.train(training_frame=trainingFrame, fold_column="pclass", data_leakage_handling="kfold",
             target_column=targetColumnName, encoded_columns=teColumns)
```

Reply:
Tested as well.